pkg/graph/resolvers/system_intake_actions.go

```diff
@@ -0,0 +1,93 @@
+package resolvers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/guregu/null"
+
+	"github.com/cmsgov/easi-app/pkg/appcontext"
+	"github.com/cmsgov/easi-app/pkg/apperrors"
+	"github.com/cmsgov/easi-app/pkg/graph/model"
+	"github.com/cmsgov/easi-app/pkg/models"
+	"github.com/cmsgov/easi-app/pkg/storage"
+)
+
+// CreateSystemIntakeActionRequestEdits creates a new action to request edits on an intake form as part of Admin Actions v2
+func CreateSystemIntakeActionRequestEdits(
+	ctx context.Context,
+	store *storage.Store,
+	fetchUserInfo func(context.Context, string) (*models.UserInfo, error),
+	input model.SystemIntakeRequestEditsInput,
+) (*models.SystemIntake, error) {
+	intake, err := store.FetchSystemIntakeByID(ctx, input.SystemIntakeID)
+	if err != nil {
+		return nil, err
+	}
+	// Set the intake current step to the requested edits step
+	formStepMap := map[model.SystemIntakeFormStep]models.SystemIntakeStep{
+		model.SystemIntakeFormStepInitialRequestForm: models.SystemIntakeStepINITIALFORM,
+		model.SystemIntakeFormStepDraftBusinessCase:  models.SystemIntakeStepDRAFTBIZCASE,
+		model.SystemIntakeFormStepFinalBusinessCase:  models.SystemIntakeStepFINALBIZCASE,
+	}
+	intake.Step = formStepMap[input.IntakeFormStep]
+	var targetForm models.GovernanceRequestFeedbackTargetForm
+	// Set the state of the requested form step and set the targeted feedback step
+	switch input.IntakeFormStep {
+	case model.SystemIntakeFormStepInitialRequestForm:
+		intake.RequestFormState = models.SIRFSEditsRequested
+		targetForm = models.GovernanceRequestFeedbackTargetIntakeRequest
+	case model.SystemIntakeFormStepDraftBusinessCase:
+		intake.DraftBusinessCaseState = models.SIRFSEditsRequested
+		targetForm = models.GovernanceRequestFeedbackTargetDraftBusinessCase
+	case model.SystemIntakeFormStepFinalBusinessCase:
+		intake.FinalBusinessCaseState = models.SIRFSEditsRequested
+		targetForm = models.GovernanceRequestFeedbackTargetFinalBusinessCase
+	default:
+		return nil, &apperrors.BadRequestError{
+			Err: fmt.Errorf("Cannot request edits on %s", input.IntakeFormStep),
+		}
+	}
+	intake, err = store.UpdateSystemIntake(ctx, intake)
+	if err != nil {
+		return nil, err
+	}
+	adminTakingAction, err := fetchUserInfo(ctx, appcontext.Principal(ctx).ID())
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Ideally we'd have all of our DB operations in a transaction to prevent errors from leaving our DB half-updated, but that's something that we'll have to figure out outside the scope of this ticket.

However, something we can do to lower the likelihood of (in this case) updating the intake but not taking the `store.CreateAction` step is to put the `fetchUserInfo` call at the top of this function. This way, the calls (like `fetchUserInfo` that are _most likely_ to intermittently fail happen before anything else, and the calls that are _least likely_ to intermittently fail are chained together (`storage` operations, which should be pretty safe)

Reply:
That's a great catch, thanks.

Reply:
resolved in [c38f771](https://github.com/CMSgov/easi-app/pull/2129/commits/c38f7719b2cd5c17d9a5d3b6c1b2479fef146da5)